ibis/backends/sql/compilers/duckdb.py

```diff
@@ -143,15 +143,17 @@ def visit_StructColumn(self, op, *, names, values):
         )
 
     def visit_ArrayDistinct(self, op, *, arg):
+        return self._array_distinct(arg)
+
+    def _array_distinct(self, arg: sge.Expression) -> sge.Expression:
+        x = sg.to_identifier("x")
+        is_null = sge.Lambda(this=x.is_(NULL), expressions=[x])
+        contains_null = self.f.list_bool_or(self.f.list_apply(arg, is_null))
         return self.if_(
             arg.is_(NULL),
             NULL,
             self.f.list_distinct(arg)
-            + self.if_(
-                self.f.list_count(arg) < self.f.len(arg),
```

Comment:
To be totally clear, the new code removes the call to `self.f.len(arg)` which is responsible for getting rid of one of the duplications.